test/integration/fake_upstream.h

```diff
@@ -161,6 +161,8 @@ class FakeUpstream : Logger::Loggable<Logger::Id::testing>, public Network::Filt
   // Network::FilterChainFactory
   bool createFilterChain(Network::Connection& connection) override;
 
+  uint32_t port() const { return socket_->localAddress()->ip()->port(); }
```

Comment:
nit: can we move this above 161 (makes it clear which functions are part of interface overrides). I'm not totally crazy that we have a port() function here either since FakeUpstream also does UDS. Can we just return localAddress maybe and then have the caller deal with the right thing to do depending on case?